components/TARGET_PSA/inc/psa/service.h

```diff
@@ -17,8 +17,10 @@
 
 #if defined(TARGET_TFM)
 #include "interface/include/psa_service.h"
-#else
+#elif defined(TARGET_MBED_SPM)
 #include "TARGET_MBED_SPM/psa_defs.h"
 #include "TARGET_MBED_SPM/COMPONENT_SPE/spm_server.h"
 #include "TARGET_MBED_SPM/COMPONENT_SPE/spm_panic.h"
+#else
+#error "Compiling psa service header on non-secure target is not allowed"
```

Comment:
Not allowed? Why?

Reply:
because you are not supposed to use psa/service/h in NSPE builds and non-PSA builds

Reply:
I would call it not expected instead of not allowed. There are no rules about what symbols can be used where.